### PR TITLE
Replace url-parse with URL

### PR DESCRIPTION
**Pull request by eatyourgreens**
Replace the [`url-parse`](https://www.npmjs.com/package/url-parse) package with the [native URL interface](https://developer.mozilla.org/en-US/docs/Web/API/URL), which is supported in Node and in browsers. Using the native interface should make both NextJS apps more secure. 

From `url-parse`'s Readme:
>`url-parse` was created in 2014 when the WHATWG URL API was not available in Node.js and the URL interface was supported only in some browsers. Today this is no longer true. The URL interface is available in all supported Node.js release lines and basically all browsers. Consider using it for better security and accuracy.

## Package
app-content-pages
app-project
lib-panoptes-js

## How to Review
Sign in and register should work in both NextJS apps. The project app should still load projects based on the slug in the page URL. 

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [x] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [x] Can submit a classification
- [x] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [x] The PR creator has described the reason for refactoring
- [x] The refactored component(s) continue to work as expected

### CODE REVIEW

File: packages/app-content-pages/src/shared/components/ZooHeaderWrapper/ZooHeaderWrapperContainer.js [hunk 1]
```diff
@@ -4,72 +4,94 @@ import { withRouter } from 'next/router'
...
   }
 
-  redirect (urlObject) {
-    const { pathname, push } = this.props.router
-    const urlString = urlObject.toString().substr(urlObject.origin.length)
-    push(pathname, urlString, { shallow: true })
+  redirect(urlObject) {
+    const { push } = this.props.router
+    const urlString = urlObject.toString()
+    push(urlString, urlString, { shallow: true })
```
Comment (eatyourgreens) [inline packages/app-content-pages/src/shared/components/ZooHeaderWrapper/ZooHeaderWrapperContainer.js:41]: I’m not 100% sure, but I think you can pass URL objects to `router.push` without converting them to strings. That would simplify the code.

**Review by mcbouslog — COMMENTED** [review]
Code changes look good, but when I run https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats locally (on Chrome) it renders "window is not defined" and there are a few Uncaught Errors, the first of which is `Uncaught Error: Target container is not a DOM element.`.

**Comment by mcbouslog** [comment]
> Code changes look good, but when I run https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats locally (on Chrome) it renders "window is not defined" and there are a few Uncaught Errors, the first of which is `Uncaught Error: Target container is not a DOM element.`.


[This Sentry issue](https://sentry.io/organizations/zooniverse-27/issues/3605940210/?environment=production&environment=%22staging%22&environment=development&environment=staging&project=1492691) is me, related to error noted.

**Comment by eatyourgreens** [comment]
I’ll stage this to kubernetes so that we can take a look. I assume the page just crashes in Node?

**Comment by eatyourgreens** [comment]
Oh, I see the error in the diff. `AuthModalContainer` is missing a couple of lines in the project app.

File: packages/app-content-pages/src/shared/components/AuthModal/AuthModalContainer.js [hunk 2]
```diff
@@ -27,53 +26,49 @@ class AuthModalContainer extends Component {
...
 
     return activeIndex
   }
 
-  getUrlObject () {
-    const { asPath } = this.props.router
-    return new Url(asPath, true)
+  getUrlObject() {
+    const isBrowser = typeof window !== 'undefined'
+    if (isBrowser) {
```
Comment (eatyourgreens) [inline packages/app-content-pages/src/shared/components/AuthModal/AuthModalContainer.js:52]: These changes also need to be added to the project app.

**Comment by eatyourgreens** [comment]
Ok, this branch is deployed here (with the error):
https://fe-project-branch.preview.zooniverse.org/projects/brooke/i-fancy-cats?env=staging

I'm going to push a change to the `AuthModalContainer` code and deploy again.

**Comment by eatyourgreens** [comment]
Re-deployed to `fe-project-branch` but it's still erroring because `url.searchParams` is undefined on the server.
https://fe-project-branch.preview.zooniverse.org/projects/marshexplorer/marsh-explorer

**Comment by eatyourgreens** [comment]
@mcbouslog I'd made changes to the auth modal in the content pages app but not copied those same changes to the project app. At the moment, the Zooniverse page header has to be edited in both places 😒 . Everything should be working now eg. https://fe-project-branch.preview.zooniverse.org/projects/marshexplorer/marsh-explorer

**Comment by eatyourgreens** [comment]
I'm also not sure why `AuthModalContainer` is rendered server-side, since you can only log in or register in the browser.

**Review by mcbouslog — APPROVED** [review]
Looks good! Tested locally, tested on https://fe-project-branch.preview.zooniverse.org/projects/marshexplorer/marsh-explorer, ran app-content locally and confirmed expected sign in and register modals appear 👍 .

**Pull request merged**